src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/script/KNNScoringUtil.java

```diff
@@ -0,0 +1,76 @@
+package com.amazon.opendistroforelasticsearch.knn.plugin.script;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+
+public class KNNScoringUtil {
+    private static Logger logger = LogManager.getLogger(KNNScoringUtil.class);
+
+    public static float l2Squared(float[] queryVector, float[] inputVector) {
+        float squaredDistance = 0;
+        for (int i = 0; i < inputVector.length; i++) {
+            float diff = queryVector[i]-inputVector[i];
+            squaredDistance += diff * diff;
+        }
+        return squaredDistance;
+    }
+
+    public static float cosinesimilOptimized(float[] queryVector, float[] inputVector, float normQueryVector) {
+        float dotProduct = 0.0f;
+        float normInputVector = 0.0f;
+        for (int i = 0; i < queryVector.length; i++) {
+            dotProduct += queryVector[i] * inputVector[i];
+            normInputVector += inputVector[i] * inputVector[i];
+        }
+        float normalizedProduct = normQueryVector * normInputVector;
+        try {
+            return (float) (dotProduct / (Math.sqrt(normalizedProduct)));
+        } catch(ArithmeticException ex) {
+            logger.debug("Possibly Division by Zero Exception. Returning min score to put this result to end. " +
+                    "Current normalized product " + normalizedProduct);
+            return Float.MIN_VALUE;
+        }
+    }
+
+    public static float cosinesimil(float[] queryVector, float[] inputVector) {
+        double dotProduct = 0.0f;
```

Comment:
even this method is not used by the plugin, the implementation can be similar to the optimized version. unnecessary casting from float to double can be removed.